### PR TITLE
Download Chinese filenames bug fix

### PR DESCRIPTION
**Pull request by adfost**
Fixes a bug where Chinese filenames would be messed up when being downloaded.

Fixes https://github.com/minio/console/issues/1351

### CODE REVIEW

File: portal-ui/src/screens/Console/Buckets/ListBuckets/Objects/ListObjects/ListObjects.tsx [hunk 1]
```diff
@@ -646,6 +647,8 @@ const ListObjects = ({
       `${bucketName}-${object.name}-${new Date().getTime()}-${Math.random()}`
     );
 
+    console.log("Downloading");
+
```
Comment (harshavardhana) [inline portal-ui/src/screens/Console/Buckets/ListBuckets/Objects/ListObjects/ListObjects.tsx:651]: do we need these logs?

File: restapi/user_objects.go [hunk 2]
```diff
@@ -395,6 +396,7 @@ func getDownloadObjectResponse(session *models.Principal, params user_api.Downlo
 				filename = prefixElements[len(prefixElements)-1]
 			}
 		}
+		escapedName := url.QueryEscape(filename)
```
Comment (harshavardhana) [inline restapi/user_objects.go:399]:
```suggestion
		escapedName := url.PathEscape(filename)
```

**Review by harshavardhana — CHANGES_REQUESTED** [review]

**Comment by jinapurapu** [comment]
"+" being appended to end of filenames on download
Screenshot from 2022-01-05 14-25-07

**Comment by jinapurapu** [comment]
Filenames being garbled if they contain "+", regardless of Chinese characters
Screenshot from 2022-01-05 14-49-36

File: portal-ui/src/screens/Console/Buckets/ListBuckets/Objects/utils.ts [hunk 3]
```diff
@@ -15,6 +15,7 @@
 // along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 import { BucketObject, RewindObject } from "./ListObjects/types";
+import { decodeFileName } from "../../../../../common/utils";
```
Comment (prakashsvmx) [inline portal-ui/src/screens/Console/Buckets/ListBuckets/Objects/utils.ts:18]: Please remove this unused import.

**Review by prakashsvmx — DISMISSED** [review]
Minor comment.

otherwise
Tested. Changes look good to me. :+1:

**Review by prakashsvmx — APPROVED** [review]

**Review by harshavardhana — APPROVED** [review]

**Comment by harshavardhana** [comment]
Strange unrelated test failure

```
=== RUN   TestGetLicenseInfoFromJWT
2022/01/06 18:19:22 failed to verify license: exp not satisfied
=== RUN   TestGetLicenseInfoFromJWT/error_because_missing_license
=== RUN   TestGetLicenseInfoFromJWT/error_because_invalid_license
2022/01/06 18:19:22 Failed to parse public key: key must be a PEM encoded PKCS1 or PKCS8 key
=== RUN   TestGetLicenseInfoFromJWT/license_successfully_verified
2022/01/06 18:19:22 failed to verify license: exp not satisfied
    config_test.go:80: GetLicenseInfoFromJWT() error = invalid license key, wantErr false
--- FAIL: TestGetLicenseInfoFromJWT (0.09s)
    --- PASS: TestGetLicenseInfoFromJWT/error_because_missing_license (0.00s)
    --- PASS: TestGetLicenseInfoFromJWT/error_because_invalid_license (0.00s)
    --- FAIL: TestGetLicenseInfoFromJWT/license_successfully_verified (0.04s)
```

**Comment by dvaldivia** [comment]
@harshavardhana that test started failing today, also on other PRs that didn't had go changes

**Comment by jinapurapu** [comment]
Filenames with "?" are being modified on download
Screenshot from 2022-01-06 11-32-06

**Comment by harshavardhana** [comment]
> TestGetLicenseInfoFromJWT

Well the stored license in the test has expired

```
{
  "alg": "ES384",
  "typ": "JWT"
}.{
  "sub": "kanagaraj+c1@minio.io",
  "cap": 50,
  "org": "Gringotts Inc.",
  "exp": 1641446169.001199,
  "plan": "STANDARD",
  "iss": "subnet@min.io",
  "aid": 1,
  "iat": 1609910169.001199
}.[Signature]
```

Expired @dvaldivia 

```
> Wed Jan 05 2022 21:16:09 GMT-0800 (Pacific Standard Time)
```

**Comment by harshavardhana** [comment]
NOTE we don't have JWT based license anymore and this test is meaningless..

**Comment by harshavardhana** [comment]
First step lets remove the test as it's not meaningful and then also remove the JWT parsing code - this is not valid anymore.

**Comment by prakashsvmx** [comment]
> Filenames with "?" are being modified on download Screenshot from 2022-01-06 11-32-06

@jinapurapu 
This is a browser limitation/bug. it would replace `<` `>` `*` `:` and few more characters if we are downloading. 

An Example:
Line 1: Original File name
Line 2 Downloaded file name

image

**Comment by harshavardhana** [comment]
Please rebase @adfost your PR.

**Review by prakashsvmx — DISMISSED** [review]
:+1:

File: portal-ui/src/screens/Console/Buckets/ListBuckets/Objects/utils.ts [hunk 4]
```diff
@@ -52,9 +51,9 @@ export const download = (
       const rspHeader = req.getResponseHeader("Content-Disposition");
 
       let filename = "download";
-
       if (rspHeader) {
-        filename = rspHeader.split('"')[1];
+        let rspHeaderDecoded = decodeURIComponent(rspHeader);
```
Comment (Alevsk) [inline portal-ui/src/screens/Console/Buckets/ListBuckets/Objects/utils.ts:55]:
```suggestion
        let rspHeaderDecoded = decodeFileName(rspHeader);
```

Please use the `decodeFileName` function located at `portal-ui/src/common/utils.ts`, that function will handle the error when a string cannot be decoded

**Review by Alevsk — CHANGES_REQUESTED** [review]

File: portal-ui/src/screens/Console/Buckets/ListBuckets/Objects/utils.ts (hunk 4 above)
Comment (adfost) [reply portal-ui/src/screens/Console/Buckets/ListBuckets/Objects/utils.ts:55]: Tried that that doesn't work

**Review by bexsoft — APPROVED** [review]
LGTM

**Review by dvaldivia — APPROVED** [review]

**Review by harshavardhana — APPROVED** [review]

**Review by Alevsk — APPROVED** [review]

**Pull request merged**